packages/next/src/shared/lib/html-context.ts

```diff
@@ -46,7 +46,20 @@ export type HtmlProps = {
   fontLoaderManifest?: FontLoaderManifest
 }
 
-export const HtmlContext = createContext<HtmlProps>(null as any)
+export const HtmlContext = createContext<HtmlProps | undefined>(undefined)
```

Comment:
Yet another case where `as any` is a Bad 😈 

Explaining: `createContext` is a hard API to get type safe. Its typings indicate that you must provide a default value for it: that way, when it's called, it's known to always return a value. (In more technical terms, `createContext<T>` takes in a parameter of type `T` so the return type is `T`, not `T | undefined`.)

Unfortunately that means that when there isn't a reasonable default value -the majority of the time in my experience 🙃- projects tend to provide something bogus like `null as any` to get around type checking. They then assume `useContext` will only be called with that context in places where someone has used that context's provider to provide a real value.

The assumption that someone's provided an overridden value isn't always correct! We see in this PR's linked bug #45024 that if someone renders `<Html>` outside of `pages/_document.(js|tsx)`, Next.js hasn't provided a context value with `HtmlContext.Provider`. So when code tries `const { ... } = useContext(HtmlContext)`, that evalues to `const { ... } = undefined`, which is a runtime error.